3rdparty/vorbis/libvorbis/wscript

```diff
@@ -0,0 +1,72 @@
+#! /usr/bin/env python
+# encoding: utf-8
+
+FRAGMENT_MEMORY_H='''#include <memory.h>
+int main (void) {
+	return 0;
+}'''
+
+FRAGMENT_MEMCMP='''#include <string.h>
+int main (void) {
+    memcmp(NULL, NULL, 0);
+    return 0;
+}'''
+
+FRAGMENT_ALLOCA_H='''#include <alloca.h>
+int main (void) {
+	int foo=10;
+	int * array = alloca(foo);
+}'''
+
+def options(opt):
+	pass
+
+def configure(conf):
+	if not conf.path.find_dir('../vorbis-src') or not conf.path.find_dir('../vorbis-src/lib'):
+		conf.fatal('Can\'t find libvorbis submodule. Run `git submodule update --init --recursive`.')
+		return
+
+	# Check for required headers
+	if conf.check_cc(fragment=FRAGMENT_MEMORY_H, msg = 'Checking for memory.h header', mandatory = False):
+		conf.define('USE_MEMORY_H', 1)
+	
+	conf.check_cc(fragmenmt=FRAGMENT_MEMCMP, msg = 'Checking for memcmp in string.h header', mandatory = False)
+	conf.check_cc(fragment=FRAGMENT_ALLOCA_H, msg = 'Checking for alloca in alloca.h header', mandatory = False)
```

Comment:
I kinda remember that we already have alloca.h check somewhere... Might be good idea to reuse, but won't be a big deal not to.

Reply:
Also, `fragmenmt` above. Looks like a typo.